packages/core/src/payment/strategies/braintree/braintree-payment-processor.ts

```diff
@@ -45,6 +45,17 @@ export default class BraintreePaymentProcessor {
     }
 
     async tokenizeCard(payment: OrderPaymentRequestBody, billingAddress: Address): Promise<NonceInstrument> {
+        const {
+            ccNumber,
+            ccExpiry,
+        } = payment.paymentData as CreditCardInstrument;
+
+        const isValidRequiredFields = !!(ccNumber && ccExpiry);
+
+        if (!isValidRequiredFields) {
+            throw new PaymentArgumentInvalidError();
+        }
+
         const requestData = this._mapToCreditCard(payment.paymentData as CreditCardInstrument, billingAddress);
```

Comment:
When you update the as on line 51 can you please adjust this too.

Reply:
@animesh1987 done